index.d.ts

```diff
@@ -0,0 +1,267 @@
+import EventEmitter from 'events';
+import { PassThrough, Writable } from 'stream';
+
+type EventsType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+  STARTED: 'started';
+  FINISHED: 'finished';
+}
+
+type zUnitOptions = Partial<{
+  defaults: unknown;
+  initial: unknown;
+  runtime: unknown;
+  bequeathed: unknown;
+}>;
+
+declare class Runnable extends EventEmitter.EventEmitter {
+  constructor(name: string);
+
+  name: string;
+
+  description: string;
+}
+
+type zUnitSyntax = {
+  describe: typeof describe;
+  xdescribe: typeof xdescribe;
+  odescribe: typeof odescribe;
+  it: typeof it;
+  xit: typeof xit;
+  oit: typeof oit;
+  before: typeof before;
+  beforeEach: typeof beforeEach;
+  after: typeof after;
+  afterEach: typeof afterEach;
+  include: typeof include;
+}
+
+type OutcomesType = {
+  PASSED: 'passed';
+  FAILED: 'failed';
+  SKIPPED: 'skipped';
+}
+
+export const Events: EventsType;
+
+export class Harness extends EventEmitter.EventEmitter {
+  constructor(testable: Testable, initial?: zUnitOptions);
+
+  numberOfTests: number;
+
+  report: unknown;
+
+  run(reporter: unknown, runtime?: zUnitOptions): Promise<unknown>;
+}
+
+export class Hook extends Runnable {
+  constructor(name: string, fn: Function, initial?: zUnitOptions);
+
+  name: string;
+
+  run(propogatedOptions: zUnitOptions): Promise<void>;
+}
+
+export class HookSet {
+  addBefores(...additions: unknown[]): this;
+
+  addAfters(...additions: unknown[]): this;
+
+  runBefores(options: zUnitOptions): Promise<void>;
+
+  runAfters(options: zUnitOptions): Promise<void>;
+}
+
+export class Options {
+  constructor(options?: zUnitOptions);
+
+  defaults: unknown;
+
+  initial: unknown;
+
+  runtime: unknown;
+
+  bequeathed: unknown;
+
+  get(name: string): unknown;
```

Comment:
get will return whatever type the named option has or undefined if not found

Reply:
Sorted